syntax = 'proto3';
package protocol;
option go_package = "github.com/CynicalCode21/amongst-bins/protocol";

// 1
message JoinGame {
	string username = 1;
}
// 2
message EchoJoinGame {
	int32 player_id = 1;
}
// 3
message PlayerPosition {
	int64 player_id = 1;
	int64 x = 2;
	int64 y = 3;
}
// 4
message ClientMovement {
	PlayerPosition player_pos = 1;
}
// 5
message ServerPositionUpdate {
	repeated PlayerPosition player_pos = 1;
}
